### ISSUE TITLE
Dasharo (coreboot+SeaBIOS) with TrenchBoot for Protectli VP4670

### ISSUE THREAD

**Issue by pietrushnic** [Dasharo/dasharo-issues#1167]
Digest https://youtu.be/RVK52BCM-ZM and provide a plan for TechPreview publication.

**Comment by pietrushnic** [comment]
@miczyg1 any chance I can get your input here?

**Comment by miczyg1** [comment]
1. If it may be part of the plan, this is something to look into:

- https://github.com/TrenchBoot/xen/issues/18 (not sure if applies to VP4670, initially observed on NVC NV4xPZ)
- https://github.com/TrenchBoot/xen/issues/16

2. Platform preparation for shipping/testing:

- Connecting a dTPM
- Provisioning the dTPM for Intel TXT (requires UEFI environment to run the provisioning tools, although 9e suite has the support for provisioning a TPM, I could not get it to work, the suite tools did not detect a TPM, despite Linux detected it...)
- Flashing the TechPreview firmware

3. Testing:

I guess we are mainly interested only in a couple of test cases:

- Installing Qubes OS
- Installing and verifying TrenchBoot works (e.g. DRTM PCR are non-zero and event log reproduces PCR values)
- Other OSes? Which ones?

4. Release publication. As usual, newsletter and docs.dasharo.com via templates.
5. Product in the shop. Some description for the shop would be needed.

**Comment by pietrushnic** [comment]
> Provisioning the dTPM for Intel TXT (requires UEFI environment to run the provisioning tools, although 9e suite has the support for provisioning a TPM, I could not get it to work, the suite tools did not detect a TPM, despite Linux detected it...)

This sounds like something that an additional package in DTS, which could be available under the Dasharo Pro Package, needs to do.

> Other OSes? Which ones?

For now, I don't think we have to extend the scope here. In the long run, maybe mainstream distros could benefit from this when upstream accepts DRTM support.

**Comment by pietrushnic** [comment]
Anyway, thanks for the input and the plan. I will try to see if I can schedule anything internally.

**Comment by miczyg1** [comment]
> This sounds like something that an additional package in DTS, which could be available under the Dasharo Pro Package, needs to do.

Technically speaking, raw commands from tpm2-tools should also work... One has to simply know what commands to invoke. It mainly requires creating TPM NV indices with correct policies (which are nearly fixed).

**Comment by pietrushnic** [comment]
@BeataZdunczyk @miczyg1 Considering this effort: https://github.com/Dasharo/dasharo-issues/issues/1269, does it make sense to release this work?

Also, let's note that if we don't release it as part of the preparation for the given conference, we will most likely not release it.

**Comment by miczyg1** [comment]
> [@BeataZdunczyk](https://github.com/BeataZdunczyk) [@miczyg1](https://github.com/miczyg1) Considering this effort: [#1269](https://github.com/Dasharo/dasharo-issues/issues/1269), does it make sense to release this work?
> 

There is no effort in https://github.com/Dasharo/dasharo-issues/issues/1269
The issue is a result of trying to use non-TXT build with AEM.

**Comment by pietrushnic** [comment]
🤔 but we plan to release Dasharo (coreboot+UEFI) with TrenchBoot support as sign of NLNet project?

**Comment by pietrushnic** [comment]
Nah, not part of MoU, but if we need to prepare firmware to get through it anyway, then why not do release which could be used for product. TBH we have to do this.

**Comment by miczyg1** [comment]
> Nah, not part of MoU, but if we need to prepare firmware to get through it anyway, then why not do release which could be used for product. TBH we have to do this.

I don't think it is a matter of "why". Rather "when". But that question should be directed elsewhere.

**Comment by pietrushnic** [comment]
@miczyg1 ping.
